GTG/core/xml.py

```diff
@@ -0,0 +1,316 @@
+            continue
+
+        except PermissionError:
+            log.debug(f'Not allowed to open: {filepath}. Trying next.')
+            continue
+
+        except etree.XMLSyntaxError as e:
+            log.debug(f'Syntax error in {filepath}. {e}. Trying next.')
+            continue
+
+    if root:
+        return root
+
+    # We couldn't open any file :(
+    else:
+        # Try making a new empty file and open it
+        try:
+
+            write_empty_file(xml_path, root_tag)
+            return open_file(xml_path, root_tag)
+
+        except IOError:
+            raise SystemError(f'Could not write a file at {xml_path}')
+
+
+def write_backups(filepath: str) -> None:
+    """Make backups for the file at filepath."""
+
+    current_back = BACKUPS
+    backup_name = get_backup_name(filepath, None)
+    backup_dir = os.path.dirname(backup_name)
+
+    # Make sure backup dir exists
+    try:
+        os.makedirs(backup_dir, exist_ok=True)
+
+    except IOError:
+        log.error(f'Backup dir {backup_dir} cannot be created!')
+        return
+
+    # Cycle backups
+    while current_back > 0:
+        older = f"{backup_name}.bak.{current_back}"
+        newer = f"{backup_name}.bak.{current_back - 1}"
+
+        if os.path.exists(newer):
+            shutil.move(newer, older)
+
+        current_back -= 1
+
+    # bak.0 is always a fresh copy of the closed file
+    # so that it's not touched in case of not opening next time
+    bak_0 = f"{backup_name}.bak.0"
+    shutil.copy(filepath, bak_0)
+
+    # Add daily backup
+    today = datetime.today().strftime('%Y-%m-%d')
+    daily_backup = f'{backup_name}.{today}.bak'
+
+    if not os.path.exists(daily_backup):
+        shutil.copy(filepath, daily_backup)
+
+
+def write_xml(filepath: str, tree: etree.ElementTree) -> None:
+    """Write an XML file."""
+
+    with open(filepath, 'wb') as stream:
+        tree.write(stream, xml_declaration=True, pretty_print=True,
+                   encoding='UTF-8')
+
+
+def create_dirs(filepath: str) -> None:
+    """Create directory tree for filepath."""
+
+    base_dir = os.path.dirname(filepath)
+    try:
+        os.makedirs(base_dir, exist_ok=True)
+    except IOError as error:
+        log.error("Error while creating directories:", error)
+
+
+def save_file(filepath: str, root: etree.ElementTree) -> None:
+    """Save an XML file."""
+
+    temp_file = filepath + '__'
+
+    if os.path.exists(filepath):
+        os.rename(filepath, temp_file)
+
+    try:
+        write_xml(filepath, root)
+
+        if os.path.exists(temp_file):
+            os.unlink(temp_file)
+
+    except (IOError, FileNotFoundError):
+        log.error(f'Could not write XML file at {filepath}')
+        create_dirs(filepath)
+
+
+def write_empty_file(filepath: str, root_tag: str) -> None:
+    """Write an empty tasks file."""
+
+    root = etree.Element(root_tag)
+    save_file(filepath, etree.ElementTree(root))
```

Comment:
So there's two ways you can go with this.

1. This way
2. A formatted string of the stub file (`f'<?xml version="1.0" encoding="UTF-8" ?>\n<{root_tag}...'`)

Option 1 will let you populate with default attributes if you want (as they'd be defined in the schema - #431).
Option 2, if you don't need attributes populated with defaults or anything, will save you some memory footprint since you don't need to instantiate an `lxml.etree._Element` object. Maybe. I'd be curious to see a benchmark between the two.